src/data/nat/digits.lean

```diff
@@ -299,6 +304,67 @@ lemma of_digits_digits_append_digits {b m n : ℕ} :
   of_digits b (digits b n ++ digits b m) = n + b ^ (digits b n).length * m:=
 by rw [of_digits_append, of_digits_digits, of_digits_digits]
 
+lemma digits_len_le_digits_len_succ (b n : ℕ) : (digits b n).length ≤ (digits b (n + 1)).length :=
+begin
+  cases b,
+  { -- base 0
+    cases n; simp },
+  { cases b,
+    { -- base 1
+      simp },
+    { -- base >= 2
+      apply nat.strong_induction_on n,
+      clear n,
+      intros n IH,
+      cases n,
+      { simp },
+      { rw [digits_add_two_add_one, digits_add_two_add_one],
+        by_cases hdvd : (b.succ.succ) ∣ (n.succ+1),
+        { rw [nat.succ_div_of_dvd hdvd, list.length_cons, list.length_cons, nat.succ_le_succ_iff],
+          apply IH,
+          exact nat.div_lt_self (by linarith) (by linarith) },
+        { rw nat.succ_div_of_not_dvd hdvd,
+          refl } } } }
+end
+
+lemma le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (digits b m).length :=
+monotone_of_monotone_nat (digits_len_le_digits_len_succ b) h
+
+-- future refactor to proof using lists (see lt_base_pow_length_digits)
+-- See review in #3498. List proof depends on : m ≠ 0 → (digits b m) ≠ 0
```

Comment:
Is `List proof depends on : m ≠ 0 → (digits b m) ≠ 0` supposed to be `List proof depends on : m ≠ 0 → (digits b m) ≠ []`?

Reply:
The latter. But I'm not particularly concerned about this, since I have a PR to handle this ready.

Reply:
the code to resolve this comment is [here](https://gist.github.com/shingtaklam1324/de250b3026e40d983d1f81ec5baef67e), and it's pretty much mathlib ready imo, so it's just waiting on this PR.